middleware/validate.js

```diff
@@ -0,0 +1,168 @@
+module.exports = {
+  validator: function (req, res, next) {
+
+    /************************
+     *  ARTICLES
+    ************************/
+    if (req.originalUrl.slice(0, 9) === '/articles') {
```

Comment:
You could use req.url and the startsWith string function to do the same thing without relying on indexes.

Reply:
Also I'm not sure if this works for the product routes now.